addons/web_editor/static/lib/odoo-editor/odoo-editor.js

```diff
@@ -0,0 +1,4632 @@
+            // except the link. Then when cliking outside the link, reset the
+            // editable zones.
+            this.automaticStepSkipStack();
+            const link = closestElement(ev.target, 'a');
+            if (link && !link.querySelector('div') && !closestElement(ev.target, '.o_not_editable')) {
+                const editableChildren = link.querySelectorAll('[contenteditable=true]');
+                this._stopContenteditable();
+                [...editableChildren, link].forEach(node => node.setAttribute('contenteditable', true));
+            } else {
+                this._activateContenteditable();
+            }
+
+            const node = ev.target;
+            // handle checkbox lists
+            if (node.tagName == 'LI' && getListMode(node.parentElement) == 'CL') {
+                if (ev.offsetX < 0) {
+                    toggleClass(node, 'o_checked');
+                    ev.preventDefault();
+                }
+            }
+        }
+
+        _onDocumentKeydown(ev) {
+            const canUndoRedo = !['INPUT', 'TEXTAREA'].includes(this.document.activeElement.tagName);
+
+            if (this.options.controlHistoryFromDocument && canUndoRedo) {
+                if (isUndo(ev) && canUndoRedo) {
+                    ev.preventDefault();
+                    this.historyUndo();
+                } else if (isRedo(ev) && canUndoRedo) {
+                    ev.preventDefault();
+                    this.historyRedo();
+                }
+            } else {
+                if (isRedo(ev) || isUndo(ev)) {
+                    this._onKeyupResetContenteditableNodes.push(
+                        ...this.editable.querySelectorAll('[contenteditable=true]'),
+                    );
+                    if (this.editable.getAttribute('contenteditable') === 'true') {
+                        this._onKeyupResetContenteditableNodes.push(this.editable);
+                    }
+
+                    for (const node of this._onKeyupResetContenteditableNodes) {
+                        this.automaticStepSkipStack();
+                        node.setAttribute('contenteditable', false);
+                    }
+                }
+            }
+        }
+
+        _onDocumentKeyup() {
+            if (this._onKeyupResetContenteditableNodes.length) {
+                for (const node of this._onKeyupResetContenteditableNodes) {
+                    this.automaticStepSkipStack();
+                    node.setAttribute('contenteditable', true);
+                }
+                this._onKeyupResetContenteditableNodes = [];
+            }
+        }
+
+        /**
+         * Prevent the pasting of HTML and paste text only instead.
+         */
+        _onPaste(ev) {
+            ev.preventDefault();
+            const pastedText = (ev.originalEvent || ev).clipboardData.getData('text/plain');
+            this._insertHTML(pastedText.replace(/\n+/g, '<br/>'));
```

Comment:
this seems to be a case where `_insertHTML()` gets passed text instead of markup. Replacing NLs with BR in a text does not make it HTML. An escaping step seems to be missing here.

Reply:
Good catch. This is fixed by https://github.com/odoo-dev/odoo-editor/pull/179. Rebuild incoming.

Reply:
@odony You are right. I have changed it to insert text rather than HTML.